---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.align = "center",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# paramlink2

<!-- badges: start -->
[![CRAN status](https://www.r-pkg.org/badges/version/paramlink2)](https://CRAN.R-project.org/package=paramlink2)
[![](https://cranlogs.r-pkg.org/badges/grand-total/paramlink2?color=yellow)](https://cran.r-project.org/package=paramlink2)
[![](https://cranlogs.r-pkg.org/badges/last-month/paramlink2?color=yellow)](https://cran.r-project.org/package=paramlink2)
<!-- badges: end -->

The goal of **paramlink2** is to perform parametric linkage analysis in medical pedigrees. Both singlepoint and multipoint LOD scores are supported (the latter requires [MERLIN](https://csg.sph.umich.edu/abecasis/merlin/)). The package includes tools for visualising LOD scores and summarising peaks for use in downstream analysis.

**paramlink2** is part of the **ped suite** collection of packages for pedigree analysis in R, featured in the book [Pedigree Analysis in R](https://www.elsevier.com/books/pedigree-analysis-in-r/vigeland/978-0-12-824430-2) (Vigeland, 2021). Chapter 9 of this book gives an introduction to linkage analysis, and includes a detailed worked example using **paramlink2**.

As hinted to by the package name, **paramlink2** is a reinstalment of [paramlink](https://CRAN.R-project.org/package=paramlink), which is no longer actively developed. The two version are not compatible, and all new users are recommended to use **paramlink2**.

## Installation
To get the current official version of **paramlink2**, install from CRAN as follows:
```{r, eval = FALSE}
install.packages("paramlink2")
```

Alternatively, you can obtain the latest development version from GitHub:
```{r, eval = FALSE}
# install.packages("devtools") # install devtools if needed
devtools::install_github("magnusdv/paramlink2")
```

## An example
```{r}
library(paramlink2)
```

The family below is affected with an autosomal dominant disorder. The built-in dataset `dominant1` contains genotypes for 14 of the members, at 248 SNP markers on chromosome 1. We will perform parametric linkage analysis on the dataset, hoping to identify a genomic region linked to the disease locus. 

```{r dominant1Ped, echo = F, fig.height = 4.5, out.width = "80%"}
plot(dominant1$ped, aff = dominant1$aff, starred = typedMembers)
```

The dataset contains three elements: `ped`, `aff` and `map`. For simplicity we store these in separate variables:
```{r}
ped = dominant1$ped
aff = dominant1$aff
map = dominant1$map
```

The pedigree plot above was produced with the command
```{r, eval = F}
plot(ped, aff = aff, starred = typedMembers)
```

In order to compute LOD scores, we need to provide a disease model. This is conveniently done with the function `diseaseModel()`. For this example we use a fully penetrant autosomal dominant model:
```{r}
modAD = diseaseModel("AD")

modAD
```
The `diseaseModel()` function contains a number optional arguments which can be used to create more realistic models, including phenocopies, reduced penetrance and liability classes. Consult the documentation `?diseaseModel` for details about these!

Singlepoint LOD scores for the family are now computed as follows:
```{r}
lods = lod(ped, aff = aff, model = modAD)
```

The `summary()` function prints the highest score:
```{r}
summary(lods)
```

Furthermore, `plot()` produces a nice graph of the LOD scores:
```{r singleLodGraph, fig.height = 3, echo = -1}
par(mar = c(5.1, 4.1, 1, 1))
plot(lods)
```

As typical for singlepoint scores, the graph is quite noisy and not easy to interpret, although we see indications of a peak. A cleaner picture may be obtained by *multipoint* analysis, which is available if you have MERLIN installed on your computer. A special wrapper, `merlinLod()` takes care of all the input and output files to MERLIN, so that we never have to leave R. Moreover, the syntax is similar to that of `lod()` except that we may add a linkage map of the markers.

```{r multiLod}
lodsM = merlinLod(ped, aff = aff, model = modAD, map = map)
```

The following command plots the MERLIN scores together with the singlepoint scores:
```{r multiLodGraph, fig.height = 3, echo = -1}
par(mar = c(5.1, 4.1, 1, 1))
plot(lods, col = 8)
points(lodsM, col = 2)
legend("topright", c("Multi", "Single"), col = c(2, 8), lwd = 2)
```

This graph shows a convincing peak of LOD = 3, which is close to the traditional significance threshold LOD = 3.3 for AD disorders. 
We print some details about the peak and its location:

```{r}
peakSummary(lodsM, threshold = 2)
```

In other words, the disease locus is most likely somewhere on chromosome 1 between 34 Mb and 56 Mb.
